visualization/app/codeCharta/core/settings/settings.service.ts

```diff
@@ -259,43 +262,45 @@ export class SettingsService implements DataServiceSubscriber, CameraChangeSubsc
     }
 
     /**
-     * @returns {number}
-     *
-     * Function that computes the margin applied to a scenario related the square root of (the area divided
+     * Computes the margin applied to a scenario related the square root of (the area divided
      * by the number of buildings)
      */
-    public computeMargin(
-        map: CodeMap = this.dataService.data.renderMap,
-        areaMetric: string = this.settings.areaMetric,
-        settingsMargin: number = this.settings.margin,
-        dynamicMargin: boolean = this.settings.dynamicMargin
-    ): number {
-
+    public computeMargin(s: Settings): number {
+        s = this.settings ? this.settings : s;
         let margin: number;
-        if (map !== null && dynamicMargin) {
-            let leaves = hierarchy<CodeMapNode>(map.nodes).leaves();
+        if (s.map !== null && s.dynamicMargin) {
+            let leaves = hierarchy<CodeMapNode>(s.map.nodes).leaves();
             let numberOfBuildings = 0;
             let totalArea = 0;
-            leaves.forEach((c: HierarchyNode<CodeMapNode>) => {
+
+            leaves.forEach((node: HierarchyNode<CodeMapNode>) => {
                 numberOfBuildings++;
-                if(c.data.attributes && c.data.attributes[areaMetric]){
-                    totalArea += c.data.attributes[areaMetric];
+                if(node.data.attributes && node.data.attributes[s.areaMetric]){
+                    totalArea += node.data.attributes[s.areaMetric];
                 }
             });
 
-            margin = SettingsService.MARGIN_FACTOR * Math.round(Math.sqrt(
-                (totalArea / numberOfBuildings)));
-
-            margin = Math.min(100,Math.max(SettingsService.MIN_MARGIN, margin));
+            margin = SettingsService.MARGIN_FACTOR * Math.round(Math.sqrt((totalArea / numberOfBuildings)));
+            return Math.min(SettingsService.MAX_MARGIN, Math.max(SettingsService.MIN_MARGIN, margin));
+        } else {
+            return this.settings ? this.settings.margin : SettingsService.MIN_MARGIN;
```

Comment:
Check if else statement and MIN_MARGIN, as margin should always be set, is necessary

Reply:
The margin has to be set at some point to be able to call it, but the `MIN_MARGIN` can also be set inside the `getDefaultSettings()`.